task_templates/pipelines/python3_anomaly_detection/custom.py

```diff
@@ -1,40 +1,112 @@
-    output_dir: the path to write output. This is the path provided in '--output' parameter of the
-        'drum fit' command.
-    row_weights: An array of non-negative numeric values which can be used to dictate how important
+    X: pd.DataFrame
+        Training data that DataRobot passes when this task is being trained. Note that both the training data AND
+        column (feature) names are passed
+    y: pd.Series
+        Project's target column. In the case of anomaly detection None is passed
+    output_dir: str
+        A path to the output folder (also provided in --output paramter of 'drum fit' command)
+        The artifact [in this example - containing the trained sklearn pipeline]
+        must be saved into this folder
+    class_order: Optional[List[str]]
+        This indicates which class DataRobot considers positive or negative. E.g. 'yes' is positive, 'no' is negative.
+        Class order will always be passed to fit by DataRobot for classification tasks,
+        and never otherwise. When models predict, they output a likelihood of one class, with a
+        value from 0 to 1. The likelihood of the other class is 1 - this likelihood.
+        The first element in the class_order list is the name of the class considered negative inside DR's project,
+        and the second is the name of the class that is considered positive
+    row_weights: Optional[np.ndarray]
+        An array of non-negative numeric values which can be used to dictate how important
         a row is. Row weights is only optionally used, and there will be no filtering for which
         custom models support this. There are two situations when values will be passed into
-        row_weights, during smart downsampling and when weights are explicitly provided by the user
-    kwargs: Added for forwards compatibility
+        row_weights, during smart downsampling and when weights are explicitly specified in the project settings
+    kwargs
+        Added for forwards compatibility
 
     Returns
     -------
-    Nothing
+    None
+        fit() doesn't return anything, but must output an artifact
+        (typically containing a trained object) into output_dir
+        so that the trained object can be used during scoring.
     """
+
     # Feel free to delete which ever one of these you aren't using
     estimator = make_anomaly()
 
     estimator.fit(X)
-    # You must serialize out your model to the output_dir given, however if you wish to change this
-    # code, you will probably have to add a load_model method to read the serialized model back in
-    # When prediction is done.
-    # Check out this doc for more information on serialization https://github.com/datarobot/custom-\
-    # model-templates/tree/master/custom_model_runner#python
-    # NOTE: We currently set a 10GB limit to the size of the serialized model
+
+    # Dump the trained object [in this example - a trained one class SVM model]
```

Comment:
In all cases use Serialize instead of Dump.  This could cause confusion for users who are not native English speakers.

Reply:
Good suggestion, I'll implement it!

Reply:
imo "dump" is better, because "serialize" is a technical term people don't know. dump is a term too, but simpler to understand because it's a commonly used word, so not native speakers have better chances guessing what it means.

Reply:
Unless you feel strongly @scottp-ml can we leave "dump" as is and potentially open an improvement ticket up later? Based on user feedback I'll imagine we'll be enhancing / improving a large chunk of our docs anyway

Reply:
I think Misha overrode my suggestion.